masonry/src/contexts.rs

```diff
@@ -577,9 +583,32 @@ impl_context_method!(
 pub struct TimerToken;
 
 impl EventCtx<'_> {
-    // TODO - Document
+    // TODO - clearly document all semantics of pointer capture when they've been decided on
```

Comment:
Another question here: Is it valid for a disabled widget to have the pointer capture?

Reply:
Currently, any widget can release the capture during any event. That's either a target of an event or one of its ancestors. The target can be the widget with the pointer capture for a pointer event, or the widget with keyboard focus for a text event, or can be the target widget of an accessibility event. Widgets are not currently notified when they lose the capture in this way.

There's no hard requirement at the moment that says, e.g., the pointer capture widget (if any) should be equal to the widget with keyboard focus (if any).

Reply:
It is at the moment. If we decide against that, the same should be invalid for keyboard focus. Capturing or requesting focus should not be allowed when disabled, and in `EventCtx::set_disabled` we should traverse the tree up from the widgets with pointer capture and keyboard focus to check whether their subtree is now disabled.